### PR TITLE
Add button to load references

### PR DESCRIPTION
**Pull request by realVinayak**
Done because main agents have a lot of references, and delete blockers query always times out (WOULD NOT be improved by https://github.com/specify/specify7/pull/4009). 

Ideally, we improve performance of delete blockers query

### CODE REVIEW

**Comment by realVinayak** [comment]
image

**Review by melton-jason — APPROVED** [review]

File: specifyweb/frontend/js_src/lib/components/Merging/Usages.tsx [hunk 1]
```diff
@@ -30,13 +31,22 @@ function Usages({
...
+    React.useCallback(
+      async () => (loadBlockers ? fetchBlockers(resource, true) : undefined),
+      [loadBlockers, resource]
+    ),
     false
   );
   return (
     <td className="h-[theme(spacing.40)] flex-col !items-start overflow-auto">
-      {blockers === undefined ? (
+      {!loadBlockers ? (
```
[the remark this answers is not in the document]
Comment (realVinayak) [reply specifyweb/frontend/js_src/lib/components/Merging/Usages.tsx:45]: I personally like pushing the important parts of the component down, and the filter conditions, and edge cases towards the up.  That way, if we are deep into the body of the function (or chain of ternaries), it becomes easier to just consider the valid cases. For example, the main part is the delete blocker component, and the button, and loading text are the uninteresting parts, which should be pushed up.

[the remark this answers is not in the document]
Comment (melton-jason) [reply specifyweb/frontend/js_src/lib/components/Merging/Usages.tsx:45]: No problem! 
The guard clause way of formatting code. Makes sense.

**Comment by grantfitzsimmons** [comment]
I LIKE IT!!!!!!!

**Comment by grantfitzsimmons** [comment]
I LOVE IT!!!!!!!!!

**Comment by carlosmbe** [comment]
Could we make it a toggle

**Comment by realVinayak** [comment]
@grantfitzsimmons thoughts? https://github.com/specify/specify7/pull/4015#issuecomment-1721865884

We can discuss that with the team, I'd vote for keeping it the way it is. I wouldn't find that particularly useful, but I also don't use specify.

**Comment by grantfitzsimmons** [comment]
I would leave it the way it is. Ideally, we solve the root issues with the performance problems, then we can remove this button entirely

**Review by maxxxxxdlp — APPROVED** [review]
besides not crashing the server, this might even improve UX a bit (in some cases only):
1. references loading unprompted may be distracting if user doesn't care about references for these records and just want to focus on the data section
2. references loading slows down the server, thus if user just wants to do merge without looking at references, they were still suffering the performance penalty

**Review by grantfitzsimmons — APPROVED** [review]

**Pull request merged**